### PR TITLE
clean generated client code and cached packages

### PR DESCRIPTION
**Pull request by MHBauer**
- clean the generated clientset, listeners & informers

in addition to the conversions deepcopies and defaults.

### CODE REVIEW

**Comment by MHBauer** [comment]
helps with #385

**Comment by MHBauer** [comment]
should fully resolve #385.

File: Makefile [hunk 1]
```diff
@@ -259,12 +259,14 @@ clean-bin:
...
 
 clean-build-image:
+	rm -rf .pkg
 	rm -f .scBuildImage
 	docker rmi -f scbuildimage > /dev/null 2>&1 || true
 
 clean-generated:
 	rm -f .generate_files
 	find $(TOP_SRC_DIRS) -name zz_generated* -exec rm {} \;
+	rm -rf pkg/client/*_generated
```
Comment (pmorie) [inline Makefile]: how about also:

`types.generated.go`
`openapi_generated.go`

Comment (MHBauer) [inline Makefile]: sure.

Comment (MHBauer) [inline Makefile]: types.generated.go needs to be done in a follow up because `build/update-codecgen.sh` freaks out if it isn't there.

Comment (pmorie) [inline Makefile]: Okay, that's cool

**Comment by duglin** [comment]
LGTM

File: Makefile [hunk 2]
```diff
@@ -259,12 +259,15 @@ clean-bin:
 	rm -f .generate_exes
 
 clean-build-image:
+	rm -rf .pkg
 	rm -f .scBuildImage
 	docker rmi -f scbuildimage > /dev/null 2>&1 || true
 
 clean-generated:
```
Comment (pmorie) [inline Makefile]: I don't think that `clean` should call `clean-generated`, actually.

**Comment by vaikas-google** [comment]
LGTM, pending the last discussion point with @pmorie. Gave the LGTM3, but don't want to merge until the clean calling clean-generated issue has been resolved. Either way is fine with me.

**Comment by pmorie** [comment]
I have no issue with the content of the clean-generated package, I would just rather call it `purge-generated` and not have it called by clean.  I would be fine if `clean-generated` did a `git checkout --` on generated files.

**Comment by pmorie** [comment]
..and I'm fine doing it in a followup, so I'll merge this now.

**Pull request merged**